roles/network_plugin/weave/tasks/seed.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: Weave seed | Set seed if first time
+  set_fact:
+    seed: '{% for host in groups["k8s-cluster"] %}{{ hostvars[host]["ansible_default_ipv4"]["macaddress"] }}{% if not loop.last %},{% endif %}{% endfor %}'
+  when: "weave_seed == 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set seed if not first time
+  set_fact:
+    seed: '{{ weave_seed }}'
+  when: "weave_seed != 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set peers if fist time
+  set_fact:
+    peers: '{{ weave_ip_current_cluster }}'
+  when: "weave_peers == 'unset'"
+  run_once: true
+  tags: confweave
+
+- name: Weave seed | Set peers if existing peers
+  set_fact:
+    peers: '{{ weave_peers }}{% if weave_ip_current_cluster not in weave_peers %} {{ weave_ip_current_cluster }}{% endif %}'
```

Comment:
You should actually set weave_ip_current_cluster to an array so you can match entire elements, rather than try to match a string within a string. Example:

```
a = ['10', '100', '1000']
'10' in a
True
b = "100 1000"
'10' in b
True
```